### PR TITLE
internal/pubsub: remove ackIDBox

### LINKED ISSUE #814: pubsub: remove ackIDBox type
@jba says we can do without it.

### PR DESCRIPTION
**Pull request by jba**
Fixes #814.

### CODE REVIEW

File: internal/pubsub/acks_test.go [hunk 1]
```diff
@@ -176,7 +176,7 @@ func TestTooManyAcksForASingleBatchGoIntoMultipleBatches(t *testing.T) {
 
 func TestAckDoesNotBlock(t *testing.T) {
 	ctx, cancel := context.WithCancel(context.Background())
-	m := &driver.Message{}
+	m := &driver.Message{AckID: 0}
```
Comment (ijt) [inline internal/pubsub/acks_test.go]: What does this have to do with removing ackIDBox?

Comment (jba) [reply internal/pubsub/acks_test.go]: The reflect package chokes on nil interface values. I don't fully understand why, but AckIDs will never be nil in practice so it shouldn't matter except for tests.

Comment (ijt) [reply internal/pubsub/acks_test.go]: Probably worth a comment.

Comment (jba) [reply internal/pubsub/acks_test.go]: Done.
